paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentPaytrailService.java

```diff
@@ -70,4 +86,119 @@ public PaymentMethodDto[] getOnlinePaymentMethodList(String merchantId, String n
+        if (userId == null || userId.isEmpty()) {
+            log.warn("creating payment without user rejected, orderId: " + orderId);
+            throw new CommonApiException(
+                    HttpStatus.FORBIDDEN,
+                    new Error("rejected-creating-payment-for-order-without-user", "rejected creating payment for order without user, order id [" + orderId + "]")
+            );
+        }
+    }
+
+    private void isValidOrderStatusToCreatePayment(String orderId, String orderStatus) {
+        // check order status, can only create payment for confirmed orders
+        if (!"confirmed".equals(orderStatus)) {
+            log.warn("creating payment for unconfirmed order rejected, orderId: " + orderId);
+            throw new CommonApiException(
+                    HttpStatus.FORBIDDEN,
+                    new Error("rejected-creating-payment-for-unconfirmed-order", "rejected creating payment for unconfirmed order, order id [" + orderId + "]")
+            );
+        }
+    }
+
+    private Payment createPayment(GetPaymentRequestDataDto dto, String type, String paymentId, String transactionId) {
+        OrderDto order = dto.getOrder().getOrder();
+        List<OrderItemDto> items = dto.getOrder().getItems();
+
+        if (items.isEmpty()) {
+            throw new IllegalArgumentException("Items cannot be empty.");
+        }
+
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd-HHmmss");
+
+        String namespace = order.getNamespace();
+
+        Payment payment = new Payment();
+        payment.setPaymentId(paymentId);
+        payment.setNamespace(order.getNamespace());
+        payment.setOrderId(order.getOrderId());
+        payment.setUserId(order.getUser());
+        payment.setPaymentMethod(dto.getPaymentMethod());
+        payment.setPaymentMethodLabel(dto.getPaymentMethodLabel());
+        payment.setTimestamp(sdf.format(timestamp));
+        payment.setAdditionalInfo("{\"payment_method\": " + dto.getPaymentMethod() + "}");
+        payment.setPaymentType(type);
+        payment.setStatus(PaymentStatus.CREATED);
+        payment.setTotalExclTax(new BigDecimal(order.getPriceNet()));
+        payment.setTaxAmount(new BigDecimal(order.getPriceVat()));
+        payment.setTotal(new BigDecimal(order.getPriceTotal()));
+        payment.setPaytrailTransactionId(transactionId);
+
+        createPayer(order, paymentId);
+
+        for (OrderItemDto item : items) {
+            createPaymentItem(item, paymentId, order.getOrderId());
+        }
+
+        paymentRepository.save(payment);
+        log.debug("created payment for namespace: " + namespace + " with paymentId: " + paymentId);
+
+        return payment;
+    }
+
+    private void createPaymentItem(OrderItemDto itemDto, String paymentId, String orderId) {
+        fi.hel.verkkokauppa.payment.model.PaymentItem item = new fi.hel.verkkokauppa.payment.model.PaymentItem();
```

Comment:
Use import to use remove need for fi.hel.verkkokauppa.payment.model.

Reply:
I used earlier both PaymentItem from paytrail-java-lib and the doman specific PaymentItem, so i had to use this fully qualified name for one of them. I'll fix this away now that there is only reference to one PaymentItem class type, well spotted!